### DISCUSSION TITLE
Adding signals to the search results

### DISCUSSION THREAD

**Discussion by PheoSoft** [godotengine/godot-proposals#9288] [Editor]
Hello,
Sorry if this is not the right place I am very new to godot.
tldr:
Long story short: I thought I deleted a signal and nothing happened, the signal I wanted to remove was other but I couldn't find in the right node. So I thought that the signals would also appear in the search menu.
Do you think it would be good to show results of the string you search in linked signals? sorry if this is posible to do now.

For example I am modifying the First 2D the game for the sake of learning. At the beginning I wrongly added twice a game_over to the hit signal in player and in player scene in Main, I deleted the player link to game over, that I didn't need in the first place and everything kept the same, I got confused and stupid me didn't look the signal in player scene in main, also thinking that the signal would have been deleted too. I search for game_over that was the function linked to an I couldn't find anything. At the same time of the remove of the signal I created a git repo and  I thought I was having some problem with the branches.
At the end I found the signal link doing a grep in the directory looking for the string game_over and I found the signal in the main scene in the player node...

**Comment by PheoSoft** [comment]
I just did a quick dirty mockup
![godot_search](https://github.com/godotengine/godot-proposals/assets/162583070/ccec44e2-0cc4-4910-ac0b-30be6c23c3cd)

**Comment by Calinou** [comment]
I recall there was an issue/proposal/pull request for enabling searching in `tscn` and `tres` file extensions, in addition to the existing `gd` and `gdshader`. However, it wasn't accepted because there's a high potential for corrupting scene files if you replace text incorrectly in a `.tscn` file. On top of that, it won't work with binary scenes/resources (`scn`/`res`).

We could make replacement operations ignore `tscn` and `tres` files to avoid this, but the behavior would be non-obvious at first.

**Reply by PheoSoft** [reply]
I understand, it is not easy and obvious to add that result to the search/replace dialog. Thanks for replying!
